### PR TITLE
[d3-dsv] Remove `undefined` from DSVRowString

### PR DESCRIPTION
**Pull request by pcclarke**
Currently, `DSVRowString` is defined as `[key in Columns]: string | undefined;` for all versions. This represents correct behavior for d3-dsv 0.4, where values returned from `parse()` can be either a string or undefined for missing fields.

However, a [change made in d3-dsv v1.1.2](https://github.com/d3/d3-dsv/releases/tag/v1.1.2) changed `parse()` such that it only returns strings. Instead of returning undefined for missing fields, empty strings are returned. The `DSVRowString` for v1, v2, and v3 should only be string, not undefined, as `parse()` should never return undefined values.

This pull request updates the `DSVRowString` for d3-dsv v1, v2, and v3 to reflect this change correctly.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [ ] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: [<<url here>>](https://github.com/d3/d3-dsv/releases/tag/v1.1.2)
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Review by Methuselah96 — APPROVED** [review]
Thanks!

File: types/d3-dsv/index.d.ts [hunk 1]
```diff
@@ -15,10 +15,9 @@
 
 /**
  * An object representing a DSV parsed row with values represented as strings.
- * When the DSV content is not well-structured and some column-values are missing, `undefined` is used as value.
  */
 export type DSVRowString<Columns extends string = string> = {
-    [key in Columns]: string | undefined;
+    [key in Columns]: string;
```
Comment (Methuselah96) [inline types/d3-dsv/index.d.ts:20]: Should `DSVRaw` be updated as well?

**Review by Methuselah96 — CHANGES_REQUESTED** [review]

File: types/d3-dsv/index.d.ts (hunk 1 above)
Comment (Methuselah96) [reply types/d3-dsv/index.d.ts:20]: Or should the `DSVRaw` type be removed altogether? I don't see any usages of it.

Comment (Methuselah96) [reply types/d3-dsv/index.d.ts:20]: At the least, it probably makes sense to remove `undefined` from it.

Comment (pcclarke) [reply types/d3-dsv/index.d.ts:20]: I can't see what the purpose of `DSVRaw` is, but as this is my first pull request here I'm being cautious with my changes. Removing `DSVRaw` altogether or the `undefined` from it are both fine by me.

Comment (Methuselah96) [reply types/d3-dsv/index.d.ts:20]: Yeah, I'm not sure what the purpose of it is either, but it seems correlated with `DSVRowString`, so just removing the `undefined` seems like the pragmatic choice, if you don't mind doing that as part of this PR.

**Review by Methuselah96 — APPROVED** [review]
Thanks!

File: types/d3-dsv/index.d.ts [hunk 2]
```diff
@@ -15,18 +15,17 @@
...
 };
 
 /**
  * An object in raw format before parsing, that is with only string values.
  * When the DSV content is not well-structured and some column-values are missing, `undefined` is used as value.
```
Comment (Methuselah96) [inline types/d3-dsv/index.d.ts:25]: Sorry, if you could cleanup the comment here as well, that would be great.
```suggestion
```

**Review by Methuselah96 — CHANGES_REQUESTED** [review]

**Review by Methuselah96 — APPROVED** [review]

**Review by gabritto — APPROVED** [review]

**Comment by Methuselah96** [comment]
Ready to merge

**Pull request merged**